### ISSUE TITLE
semicolons rule detects unnecessary semis when in fact they are necessary.

### ISSUE THREAD

**Issue by trusktr** [palantir/tslint#2564]
### Bug Report

- __TSLint version__: 5.1.0
- __TypeScript version__: 2.2.1
- __Running TSLint via__: tslint-loader

#### TypeScript code being linted

```ts
switch((child as any).props['data-slot']) {
    case 'header':  header  = child; break
    case 'sidebar': sidebar = child; break
    case 'track':   track   = child; break
}
```

with `tslint.json` configuration:

```json
"semicolons": [true, "never"]
```

#### Actual behavior

```
ERROR in ./Timeline.tsx
/path/to/Timeline.tsx
Unnecessary semicolon (semicolon)
  14 |         React.Children.forEach(this.props.children, child => {
  15 |             switch((child as any).props['data-slot']) {
> 16 |                 case 'header':  header  = child; break
     |                                               ^
  17 |                 case 'sidebar': sidebar = child; break
  18 |                 case 'track':   track   = child; break
  19 |             }
```

#### Expected behavior

It should not complain about that.

As a workaround, I can either put the break on a new line, or just set semicolons to false to disable checking.

**Comment by trusktr** [comment]
Or, does it just not want semis ever, even between statements on the same line?

Also the auto fixer will remove the semicolons and leave the code broken.

**Comment by ajafff** [comment]
> Or, does it just not want semis ever, even between statements on the same line?

The rule tries to detect where ASI needs a semicolon. The behavior you are seeing is a bug.
